client/containers/Pub/PubDocument/PubFileImport/MetadataEditor.js

```diff
@@ -0,0 +1,275 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { Button, Checkbox, InputGroup, TextArea } from '@blueprintjs/core';
+
+import { apiFetch } from 'utils';
+import { usePageContext } from 'utils/hooks';
+import { Avatar } from 'components';
+import { MenuButton, MenuItem } from 'components/Menu';
+import { profileUrl } from 'shared/utils/canonicalUrls';
+import { usePubContext } from '../../pubHooks';
+
+require('./metadataEditor.scss');
+
+const attributionShape = PropTypes.shape({
+	name: PropTypes.string,
+	users: PropTypes.arrayOf(PropTypes.shape({})),
+});
+
+const propTypes = {
+	onSetMetadataUpdater: PropTypes.func.isRequired,
+	proposedMetadata: PropTypes.shape({
+		attributions: PropTypes.arrayOf(attributionShape),
+	}).isRequired,
+};
+
+const guessInitials = (name) => {
+	const firstName = name
+		.split(' ')
+		.slice(0, -1)
+		.join(' ');
+	const lastName = name.split(' ').pop();
+	return firstName.charAt(0) + lastName.charAt(0);
+};
+
+const ProposedAttribution = ({ attribution, onUpdateAttribution }) => {
+	const { name, users, matchedUser, ignored } = attribution;
+
+	const handleToggleIgnored = () => {
+		if (ignored) {
+			onUpdateAttribution({ ignored: false });
+		} else {
+			onUpdateAttribution({ ignored: true, matchedUser: null });
+		}
+	};
+
+	const rendermatchedUsersMenu = () => {
+		if (!users || users.length === 0 || ignored) {
+			return null;
+		}
+
+		if (matchedUser) {
+			return (
+				<Button small minimal onClick={() => onUpdateAttribution({ matchedUser: null })}>
+					Unmatch user
+				</Button>
+			);
+		}
+
+		return (
+			<MenuButton
+				aria-label="Match to PubPub user"
+				buttonProps={{
+					minimal: true,
+					small: true,
+					rightIcon: 'chevron-down',
+					icon: (
+						<span className="matched-users-avatars">
+							{users.slice(0, 3).map((user) => (
+								<Avatar
+									width={20}
+									avatar={user.avatar}
+									borderColor="white"
+									doesOverlap={true}
+								/>
+							))}
+						</span>
+					),
+				}}
+				buttonContent={`${users.length} matching ${users.length === 1 ? 'user' : 'users'}`}
+				placement="bottom-end"
+			>
+				{users.map((user) => (
+					<MenuItem
+						text={user.fullName}
+						icon={<Avatar width={20} avatar={user.avatar} />}
```

Comment:
Ideally we would also pass in user.initials to `<Avatar />` in case the user doesn't have an image uploaded.